### ISSUE TITLE
Handshake should not modify caller-supplied headers

### ISSUE THREAD

**Issue by bradleybuda** [imanel/websocket-ruby#37]
I'm using `websocket-eventmachine-client` to connect to a server that requires the HTTP Authorization header. My code (in simplified form) looks like this:

```
uri = "wss://example.com/"
headers = { "Authorization" => "Bearer 123456" }

loop do
  EventMachine.run do
    client = WebSocket::EventMachine::Client.connect(:uri => uri, :headers => headers)
    client.disconnected do { EventMachine.stop }
  end

  sleep 5.0
end
```

This works fine, and the server is able to authenticate my client. However, if I try to reconnect when the connection is dropped, the connection fails because my `headers` hash has been modified to include the headers sent by the server during the handshake negotiation - see https://github.com/imanel/websocket-ruby/blob/e9ceb235ffcee6c0d08fe4289c998f014eef670a/lib/websocket/handshake/base.rb#L109

This was unexpected behavior to me, that the web socket client would modify a caller-provided argument, and it took a lot of debugging to track down. The workaround is simple - I can move the initialization of headers inside the loop or call `headers.dup` - but IMHO the library should be changed to not modify the headers passed in, to make it less likely that others will experience this issue. Thanks and let me know if I can answer any other questions.

**Comment by imanel** [comment]
That's definitely bug in my library. Will fix it in a minute and release new version.

**Comment by bradleybuda** [comment]
Yes - that fixed it! Thank you for the quick reply and the very useful gems.
